telemetry/csharp/ToolkitTelemetryGenerator/Program.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using CommandLine;
+using ToolkitTelemetryGenerator.Models;
+
+namespace ToolkitTelemetryGenerator
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Options options = null;
+            Parser.Default.ParseArguments<Options>(args)
+                .WithParsed<Options>(o =>
+                {
+                    if (string.IsNullOrWhiteSpace(o.OutputFolder))
+                    {
+                        o.OutputFolder = Directory.GetCurrentDirectory();
+                    }
+
+                    options = o;
+                });
+
+            if (options == null)
+            {
+                throw new Exception("Program Options are undefined or missing.");
```

Comment:
Can the library actually throw up like this? how would options ever be null here?

Reply:
Defensive programming